### ISSUE TITLE
[2.0.0]Registry functionality missing from 1.3.4

### ISSUE THREAD

**Issue by eclipse88** [phalcon/cphalcon#10115]
Not supporting this very useful functionality from 1.3.4
```php
$di->set('registry', function() {
    return new Phalcon\Registry();
}, true);

//not working in 2.0.0
$this->registry->user = [];
$this->registry->user['id'] = 1;

//must doing this
$this->registry->user = ['id' => 1];
```
and if you collecting data from entire application you need to do this:
``` php
$data = [];
if($this->registry->offsetExists('data'){//by the way isset($this->registry->data) always return false in 2.0.0
     $data = $this->registry->data;
}
$data['asd'] = 'asd';
$this->registry->data = $data;

//instead
$this->registry->data['asd'];
```

**Comment by KorsaR-ZN** [comment]
@eclipse88 
I corrected only part of the problem.
The rest of the fix is not possible, because Marshmallow not support links!

It works
```
if ( $this->registry->data) {
   ...
}
```

**Comment by andresgutierrez** [comment]
@KorsaR-ZN Marshmallow ?

**Comment by sergeyklay** [comment]
@KorsaR-ZN Marshmallow LOL :smile_cat: 

@andresgutierrez Marshmallow = Transliterate Zephir  to Russian and translate to English
![mallow-1](https://cloud.githubusercontent.com/assets/1256298/7294321/d461c7dc-e9b4-11e4-891c-b324988e800e.jpg)

**Comment by KorsaR-ZN** [comment]
@andresgutierrez, @sergeyklay :)
Now this is a new name for Zephir :D

**Comment by Green-Cat** [comment]
@andresgutierrez fixed in 2.0.1
